File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-6.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = dict()
        maxLen = 0
        l = 0

        for r in range(len(s)):
            count[s[r]] = count.get(s[r], 0) + 1
            maxf = max(count.values())

            lenLeft = (r - l + 1) - maxf

            while lenLeft > k:
                count[s[l]] -= 1
                l += 1
                maxf = max(count.values())
                lenLeft = (r - l + 1) - maxf

            maxLen = max(maxLen, r - l + 1)

        return maxLen

File: Data Structures & Algorithms/is-anagram/submission-6.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length !== t.length) return false
        let freq_s = new Map()
        for (const c of s){
            freq_s.set(c, (freq_s.get(c) || 0) + 1)
        }
        for (const c of t){
            let count = freq_s.get(c) || 0
            if (count === 0) return false
            freq_s.set(c, count - 1)
        }

        return true
    }
}
